### DISCUSSION TITLE
How does NVDA determine whether an application supports Browse and Focus Modes?

### DISCUSSION THREAD

**Discussion by yolpsoftware** [nvaccess/nvda#13448] [Q&A]
When running NVDA on Windows, Browse and Focus Mode is supported on some applications: Chrome, Firefox, Word, VS Code. In those applications, I can use Insert+Space to switch between the modes.

In other applications, like Calculator and Windows Explorer, pressing Insert+Space does not have any effect. So it seems, those applications do not support Browse Mode (which kinda makes sense, as there is not much to "read" in those applications). I assume that in those applications, the default is something like Focus Mode.

How does NVDA determine whether an application supports Browse Mode?

I'm asking because we have a Windows application which contains a browser, and we would like to make sure that the browser in that application supports Browse Mode. Currently, it does not.

Thank you!

**Comment by feerrenrut** [comment]
It looks like you've had an answer here: https://github.com/nvaccess/nvda/issues/13493

Browse mode is designed for "documents" rather than "applications", however applications may have embedded documents, where browse mode would be appropriate.

**Comment by cary-rowen** [comment]
btw, Can NVDA add a command to quickly jump to embedded WebView?

**Reply by feerrenrut** [reply]
@cary-rowen I'm not sure. This would require technical investigation to determine feasibility, and some discussion on use-cases to determine priority. Please start a new discussion or open an issue, ideally supplying technical information, use-cases or both.
